src/commands/commerce/store/display.ts

```diff
@@ -77,19 +80,35 @@ export class StoreDisplay extends SfdxCommand {
         if (dInfo) return dInfo as string;
         if (!this.flags.urlpathprefix && (await this.statusFileManager.getValue('urlpathprefix')))
             this.flags.urlpathprefix = await this.statusFileManager.getValue('urlpathprefix');
-        const urlpathprefix: string = this.flags.urlpathprefix
+        let urlpathprefix: string = this.flags.urlpathprefix
             ? (this.flags.urlpathprefix as string)
             : (this.flags['store-name'] as string).replace(/[\\W_]+/g, '');
-        const domainInfo = forceDataSoql(
+        let domainInfo = forceDataSoql(
             `SELECT Domain.Domain FROM DomainSite WHERE PathPrefix='/${urlpathprefix}' limit 1`,
             this.org.getUsername()
         );
         if (
             !domainInfo.result.records ||
             domainInfo.result.records.length === 0 ||
             !domainInfo.result.records[0]['Domain']
-        )
-            throw new SfdxError(messages.getMessage('view.info.noStoreMatch', [this.flags['store-name']]));
+        ) {
+            // For B2B Templates, Site.com is used which means SITE_COM_APPENDED_PATH will be appended.
+            domainInfo = forceDataSoql(
+                `SELECT Domain.Domain FROM DomainSite WHERE PathPrefix='/${urlpathprefix}${SITE_COM_APPENDED_PATH}' limit 1`,
```

Comment:
this will break B2C store as '/s' is not applicable?

Reply:
It wouldn't because the b2c store would pass the first query. I added this query in case the first original query doesn't return any results.

Reply:
would there ever be a case where both queries return something?

Reply:
No because the second query with the site_com_append wouldn't run if the first query have returned the domain.

Reply:
ah.. so this is an additional query.. It is a hacky solution